### PR TITLE
IPFS/Collections multi-platform image upload fixes

### LINKED ISSUE #959: IPFS/Marketplace image uploads for samples in WebGL & other platforms
**Describe the bug**
Right now iPFS samples works for the editor and iOS build but we need to implement them to work in WebGL

**Steps**
1. Set your bucket ID and API key in IPFS calls (SampleMain scene)
2. Build with WebGL
3. Use the scene [SampleLogin or SampleLogin - Web3Auth]
4. Login with [Web3Auth or Remote wallet]
5. Click on each of the IPFS samples

**Expected behavior**

- Upload Image works
- Upload Metadata works
- Upload Image & Metadata works

**Desktop (please complete the following information):**
 - Chainsafe SDK version: [dev]
 - Platform [WebGL]
 - Unity Version: [e.g., 2022.1]
 - Operating System: [Mac]

### PR DESCRIPTION
**Pull request by sneakzttv**
Added image upload methods in for each platform as they're all specific.
Cleaned up some code that was blocking desktop builds.

Awaiting CORS issue resolution tomorrow.

closes #959 finally

### CODE REVIEW

**Comment by sneakzttv** [comment]
webgl wont work unfortunately as storage api team wont allow cross origin :(

**Comment by kantagara** [comment]
But the cors issue should be solved by unity. There is a guide on how to solve that in our Slack (if its the same thing as for Ramp)

File: src/UnitySampleProject/Assets/Samples/web3.unity SDK/2.5.0/Web3.Unity Samples/Scenes/SampleMain.unity [hunk 1]
```diff
@@ -15427,8 +15639,8 @@ MonoBehaviour:
   m_Script: {fileID: 11500000, guid: 9a86b695c20675b43ba7a0f0a496749c, type: 3}
   m_Name: 
   m_EditorClassIdentifier: 
-  apiSecretKey: Fill In Your API Secret Key From Storage
-  bucketId: Fill In Your Bucket ID From Storage
+  apiSecretKey: eyJhbGciOiJFUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE3MTY0MDA1MzEsImNuZiI6eyJqa3UiOiIvY2VydHMiLCJraWQiOiI5aHE4bnlVUWdMb29ER2l6VnI5SEJtOFIxVEwxS0JKSFlNRUtTRXh4eGtLcCJ9LCJ0eXBlIjoiYXBpX3NlY3JldCIsImlkIjo5NDc2LCJ1dWlkIjoiMmEyYWY0MWYtMTQ4MC00YWE0LWE1MDYtMjJkNjA3OTUzNjU4IiwicGVybSI6eyJiaWxsaW5nIjoiKiIsInNlYXJjaCI6IioiLCJzdG9yYWdlIjoiKiIsInVzZXIiOiIqIn0sImFwaV9rZXkiOiJWTUFLUEZXSlpPSlBZSE5LSllYTSIsInNlcnZpY2UiOiJzdG9yYWdlIiwicHJvdmlkZXIiOiIifQ.he_5VYdwIItgAICTLT5mxAxJmY--LPAEyVAGGZfy2YW8h4aYtVcZy5XIDm0gAC6bYQYC1K4aqLstNYOF9hNVRQ
```
Comment (juans-chainsafe) [inline src/UnitySampleProject/Assets/Samples/web3.unity SDK/2.5.0/Web3.Unity Samples/Scenes/SampleMain.unity:15642]: I believe the values here need to be removed

**Review by juans-chainsafe — CHANGES_REQUESTED** [review]
Im getting this error when trying to build in Desktop:
```
/Users/juanmanuelspoleti/Desktop/workspace/web3.unity/Packages/io.chainsafe.web3-unity/Runtime/Scripts/EVM/IPFS/UploadPlatforms.cs(37,31): error CS0120: An object reference is required for the non-static field, method, or property 'UploadPlatforms.UploadImageOsx()'
```

WebGL build is working, we have the known issue with CORS (@sneakzttv are you going to try the Ramp solution that Nikola mention?).

Editor is working.

iOS build is working in Unity, but when you try to build in XCode, I see this error that looks related to IPFS:
Screenshot 2024-05-24 at 9 27 30 AM

**Comment by juans-chainsafe** [comment]
> Im getting this error when trying to build in Desktop:
> 
> ```
> /Users/juanmanuelspoleti/Desktop/workspace/web3.unity/Packages/io.chainsafe.web3-unity/Runtime/Scripts/EVM/IPFS/UploadPlatforms.cs(37,31): error CS0120: An object reference is required for the non-static field, method, or property 'UploadPlatforms.UploadImageOsx()'
> ```
> 
> WebGL build is working, we have the known issue with CORS (@sneakzttv are you going to try the Ramp solution that Nikola mention?).
> 
> Editor is working.
> 
> iOS build is working in Unity, but when you try to build in XCode, I see this error that looks related to IPFS: Screenshot 2024-05-24 at 9 27 30 AM

for iOS, I also see this message that disappears, so I didnt catch it the first time:
`/Users/juanmanuelspoleti/Desktop/workspace/web3.unity/src/UnitySampleProject/Builds/Libraries/io.chainsafe.web3-unity/Runtime/Plugins/FileUploader/FilePicker.mm:1:12 Cocoa is not available when building for iOS. Consider using `#if __has_include(<Cocoa/Cocoa.h>)` to conditionally import this framework.`

File: src/UnitySampleProject/Assets/Samples/web3.unity SDK/2.5.0/Web3.Unity Samples/Scenes/SampleMain.unity (hunk 1 above)
Comment (sneakzttv) [reply src/UnitySampleProject/Assets/Samples/web3.unity SDK/2.5.0/Web3.Unity Samples/Scenes/SampleMain.unity]: good thing i clear my keys before upload xD

**Comment by sneakzttv** [comment]
Errors are fixed, just a lib issue mac side remaining, will create a separate issue for it and merge so we can fix builds for now

https://github.com/ChainSafe/web3.unity/issues/974

**Review by juans-chainsafe — APPROVED** [review]
webGL working now, and desktop too.

Only iOS is failing: https://github.com/ChainSafe/web3.unity/issues/974

**Pull request merged**